src/pipx/venv.py

```diff
@@ -439,10 +504,16 @@ def has_package(self, package_name: str) -> bool:
         return bool(list(Distribution.discover(name=package_name, path=[str(get_site_packages(self.python_path))])))
 
     def upgrade_package_no_metadata(self, package_name: str, pip_args: List[str]) -> None:
-        logger.info("Upgrading %s", package_descr := full_package_description(package_name, package_name))
+        logger.info(
+            "Upgrading %s",
+            package_descr := full_package_description(package_name, package_name),
+        )
         with animate(f"upgrading {package_descr}", self.do_animation):
-            pip_process = self._run_pip(["--no-input", "install"] + pip_args + ["--upgrade", package_name])
-        subprocess_post_check(pip_process)
+            if self.installer != "uv":
+                upgrade_process = self._run_pip(["--no-input", "install"] + pip_args + ["--upgrade", package_name])
+            else:
+                upgrade_process = self._run_uv(["install", "--upgrade", package_name])
```

Comment:
Creating a generic function `Venv._run_installer()` selecting the right command according to the installer value might simplify this and other logic?